engine/apps/grafana_plugin/helpers/client.py

```diff
@@ -409,10 +404,17 @@ def get_instances(self, query: str, page_size=None):
                 yield page
                 cursor = page["nextCursor"]
 
+    def _is_stack_in_certain_state(self, stack_id: str, state: str) -> bool:
+        instance_info = self.get_instance_info(stack_id)
+        if not instance_info:
+            return False
+        return instance_info.get("status") == state
+
     def is_stack_deleted(self, stack_id: str) -> bool:
-        url = f"instances?includeDeleted=true&id={stack_id}"
-        instance_infos, _ = self.api_get(url)
-        return instance_infos["items"] and instance_infos["items"][0].get("status") == self.STACK_STATUS_DELETED
```

Comment:
refactoring to use `GET /instances/{stack_id}` like we were already doing in `is_rbac_enabled_for_stack` (`status` attribute is also by the underlying endpoint used there)

Reply:
is_stack_deleted needs to keep query parameter for includeDeleted=true otherwise it will not find anything

Reply:
good point, reverted change to `GcomAPIClient.is_stack_deleted` for now 👍 

(tested `GET /instances/<stackid>` call with `?includeDeleted=true` and it returns the same results as calling `GET /instances` with that query param.. won't include it in this PR as there are other calls to `GcomAPIClient.get_instance_info` and I'm not 100% sure if there would be unintended consequences elsewhere)